var/spack/repos/builtin/packages/motioncor2/package.py

```diff
@@ -41,18 +42,22 @@ class Motioncor2(Package):
         deprecated=True,
     )
 
+    depends_on("cuda@10.2,11.1:11.8,12.1", type="run")
+    depends_on("libtiff", type="run")
+
     def url_for_version(self, version):
         return "file://{0}/MotionCor2_{1}.zip".format(os.getcwd(), version)
 
-    depends_on("cuda@10.2,11.1:11.5", type="run")
-    depends_on("libtiff", type="run")
+    def setup_run_environment(self, env):
+        # As this is downloaded precompiled, we can't set library paths when building and
+        # have to manually specify the location of the CUDA libs (requires libcufft.so)
+        env.prepend_path("LD_LIBRARY_PATH", self.spec["cuda"].prefix.lib64)
```

Comment:
Maybe weird idea, but can we try to check if we can use `patchelf` in a post-install step to add an RPATH?

Reply:
Yep, that installs and tests OK. Updates in new push...